src/core/src/bootstrap/EnvLayer.py

```diff
@@ -316,11 +316,15 @@ def read_with_retry(self, file_path_or_handle):
             # only fully emulate non_exclusive_files from the real recording; exclusive files can be redirected and handled in emulator scenarios
             if not self.__emulator_enabled or (isinstance(file_path_or_handle, str) and os.path.basename(file_path_or_handle) not in self.__non_exclusive_files):
                 file_handle, was_path = self.__obtain_file_handle(file_path_or_handle, 'r')
-                value = file_handle.read()
-                if was_path:  # what was passed in was not a file handle, so close the handle that was init here
-                    file_handle.close()
-                self.__write_record(operation, code=0, output=value, delay=0)
-                return value
+                try:
+                    value = file_handle.read()
+                    if was_path:  # what was passed in was not a file handle, so close the handle that was init here
+                        file_handle.close()
+                    self.__write_record(operation, code=0, output=value, delay=0)
+                    return value
+                except Exception as error:
+                    print("Unable to read from {0} (retries exhausted). Error: {1}.".format(str(file_path_or_handle), repr(error)))
+                    return None
```

Comment:
This function is not meant to swallow exceptions like this. The caller must be prepared to handle exceptions on their own. What this function is supposed to do, however, is to actually do retries for MAX_FILE_OPERATION_RETRY_COUNT before raising the exception up. Replicate the retry and exception handling+raising logic as in write_with_retry. What caller was affected by this? Additional comments can be provided based on that.

Reply:
ToDo

Reply:
Resolved. Updated the function per feedback and modified places where this was invoked to handle any raised exceptions. We already were doing that in old code, just one new code path added in this PR was not handling the exception: __get_current_auto_os_updates_setting_on_machine() in AptitudePackageManager